content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2290.mdx

```diff
@@ -0,0 +1,194 @@
+    In this example, the highlighted constraints did not conform to Prisma's default naming convention and now include the `map` attribute field:
+
+    ```prisma highlight=11;normal
+    model User {
+      id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+1.  Optionally introspect your **production database** and double-check that constraint names in production and development are in sync. If you have external applications or scripts that rely on a specific constraint name in production, make sure that these constraint names are represented in the Prisma schema.
+
+    In the following example, the `Post` model has a foreign key constraint with a custom name in production that does not match development:
+
+      <TabbedContent tabs={[<FileWithIcon text="Development environment" icon="prisma"/>, <FileWithIcon text="Production environment" icon="prisma"/>]}>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+      </tab>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Production_Name")
+    }
+    ```
+
+      </tab>
+
+      </TabbedContent>
+
+    To synchronize constraint names between development and production, make sure that the constraint name in your Prisma schema is `map: "Custom_Production_Name"`.
+
+1.  Optionally rename any explicitly defined constraint names where you do not want to use Prisma's naming convention, but do want to use a different constraint name:
+
+    ```prisma highlight=2;delete|3;add
+    model User {
+      id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+      id    Int    @id(map: "User_ID_Primary_Key") @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+    ```
+
+1.  Optionally _remove_ the `map` field from any constraint or index attribute where you want to use the Prisma default:
+
+    ```prisma highlight=5;delete|6;add
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+      author User? @relation(fields: [authorName], references: [name])
+    }
+    ```
```

Comment:
I think the order of these two could be switched? Easier to first explain how to fall back to the default, and then how to change it to another custom one I think.